File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    let paintCount = 0;
    let currentIndex = 0;
    
    while (currentIndex < section.length) {
        // 현재 롤러로 페인트칠을 시작할 위치
        let startPosition = section[currentIndex];
        
        // 롤러로 칠할 수 있는 마지막 위치
        let endPosition = startPosition + m - 1;
        
        // 롤러를 한번 사용했으므로 카운트를 증가시킴
        paintCount++;
        
        // 롤러가 덮을 수 있는 범위 내에서 다음 페인트칠할 구역을 찾음
        while (currentIndex < section.length && section[currentIndex] <= endPosition) {
            currentIndex++;
        }
    }
    
    return paintCount;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    paint_count = 0
    current_index = 0

    while current_index < len(section):
        # 현재 페인트칠 시작 위치
        start_position = section[current_index]

        # 칠할 수 있는 마지막 위치
        end_position = start_position + m - 1

        # 롤러사용 카운트 증가
        paint_count += 1

        # 롤러가 덮을 수 있는 범위 내 다음 페인트칠할 구역 찾기
        while current_index < len(section) and section[current_index] <= end_position:
            current_index += 1

    return paint_count